### PR TITLE
Fixes an issue with search indexing and timezones

### LINKED ISSUE #59: Search indexer stops working after changing module settings
## Describe the bug
The module uses SearchModuleBase for indexing the announcements, after changing any module setting, the indexer stops working in certain timezone configurations. Dnn is not known for using best practices on timezones and in some places it uses utc, some other places  the web server timezone, and some other places the sql server timezone. This is very inconsistent.

Best practices is for the DataStore to always use UTC when saving information and for the code to handle timezones as appropriate. The Announcements module just uses the server timezone for all dates and times, this is wrong as the server might not even be in the site timezone, etc.

## Software Versions
- DNN: any
- Module: 07.02.03 and all previous.

## To Reproduce
1. Have a server that is in any negative timezone (in my case GMT-5)
2. Clear Dnn Search and run the indexer scheduled task.
3. Go to the module settings and enable RSS Feed. (changing a module setting updates the Module LastContentModifiedOnDate value for the module) which is wrong, we did not change any content by changing settings.
4. Add a new announcement
5. Run the indexer scheduled task again
6. View the RSS feed

## Wrong behaviour
The newly added announcement does not show and you can reindex all you want, it will never get indexed because changing the setting told the Dnn indexer that this point in time was the last content change, adding an annoucement does not change that and Dnn skips indexing the module until a new settings change is done, but then the schduler last indexed date is farther in the future than the last updated date of the actual announcement and well it is skipped forever.

## Expected behaviour
There are a couple of things.
1. The annoucement dates and times should not change if the webserver location or the sql server location changes, for this we need to save the data as UTC and in code convert them back to the site preferred timezone, that will handle server moves scenarios.
2. We should not call SynchronizeModule when saving settings, this method was meant for content changes, not for settings changes. Also, I believe this was there in a time where changing settings would not clear cache. Dnn now handles that, so we can simply stop calling that method.

## Error log
**Note: Debug DLL's**
This issue was not throwing any errors.

## Additional context
I am submitting a PR for this shortly and due to the nature of the changes, I will publish a pre-release for testing. 

I am specially interested in people testing this new release (7.2.4) on upgrades and localized installation usages. If you can test that new release and find issues with it, please comment...

### PR DESCRIPTION
**Pull request by valadas**
### Description of PR...
Fixes #59 and prevents marking the module last modified date when only changing settings.

## Changes made
- Ensures all DataStore usage of times are in UTC 
- Prevents marking the module last modified date when only changing settings

## PR Template Checklist

- [x] Fixes Bug
- [ ] Feature solution
- [ ] Other

### CODE REVIEW

File: AnnouncementsEdit.ascx.cs [hunk 1]
```diff
@@ -198,20 +199,29 @@ private void BindForm()
...
 
-                    ctlAudit.CreatedDate = Model.AnnouncementInfo.CreatedOnDate.ToString(CultureInfo.InvariantCulture);
+                    var user = UserController.Instance.GetCurrentUserInfo();
+                    var userPreferredCulture = CultureInfo.InvariantCulture;
+                    if (user != null && !string.IsNullOrWhiteSpace(user.Profile.PreferredLocale))
+                    {
+                        userPreferredCulture = new CultureInfo(user.Profile.PreferredLocale);
+                    }
+                    
+                    ctlAudit.CreatedDate = TimeZoneInfo.ConvertTimeFromUtc(Model.AnnouncementInfo.CreatedOnDate, ModuleContext.PortalSettings.TimeZone).ToString(userPreferredCulture);
```
Comment (mitchelsellers) [inline AnnouncementsEdit.ascx.cs:221]: I think this should be utc or portal time for the audit

Comment (valadas) [reply AnnouncementsEdit.ascx.cs:221]: Yep, it is in the portal preferred timezone but formatted in the current user culture, makes sense or I misenterpreted your comment ?

File: AnnouncementsEdit.ascx.cs [hunk 2]
```diff
@@ -198,20 +199,29 @@ private void BindForm()
                     if ((!Null.IsNull(Model.AnnouncementInfo.PublishDate)) &&
                         (Model.AnnouncementInfo.PublishDate != (DateTime)SqlDateTime.Null))
                     {
-                        publishDate.SelectedDate = Model.AnnouncementInfo.PublishDate;
-                        publishTime.SelectedDate = Model.AnnouncementInfo.PublishDate;
+                        var portalDatetime = TimeZoneInfo.ConvertTimeFromUtc(Model.AnnouncementInfo.PublishDate.Value, ModuleContext.PortalSettings.TimeZone);
```
Comment (david-poindexter) [inline AnnouncementsEdit.ascx.cs:202]:
```suggestion
                        var portalDateTime = TimeZoneInfo.ConvertTimeFromUtc(Model.AnnouncementInfo.PublishDate.Value, ModuleContext.PortalSettings.TimeZone);
```

File: AnnouncementsEdit.ascx.cs [hunk 3]
```diff
@@ -198,20 +199,29 @@ private void BindForm()
...
+                        publishDate.SelectedDate = portalDatetime;
```
Comment (david-poindexter) [inline AnnouncementsEdit.ascx.cs:203]:
```suggestion
                        publishDate.SelectedDate = portalDateTime;
```

File: AnnouncementsEdit.ascx.cs [hunk 4]
```diff
@@ -198,20 +199,29 @@ private void BindForm()
...
+                        publishTime.SelectedDate = portalDatetime;
```
Comment (david-poindexter) [inline AnnouncementsEdit.ascx.cs:204]:
```suggestion
                        publishTime.SelectedDate = portalDateTime;
```

File: Components/Business/AnnouncementsController.cs [hunk 5]
```diff
@@ -255,7 +278,7 @@ public override IList<SearchDocument> GetModifiedSearchDocuments(ModuleInfo modu
         {
             var moduleSettings = moduleInfo.ModuleSettings;
             int descriptionLenght = 100;
```
Comment (david-poindexter) [inline Components/Business/AnnouncementsController.cs:280]:
```suggestion
            int descriptionLength = 100;
```

File: Components/Business/AnnouncementsController.cs [hunk 6]
```diff
@@ -255,7 +278,7 @@ public override IList<SearchDocument> GetModifiedSearchDocuments(ModuleInfo modu
...
-            if (moduleSettings["descriptionLength"].ToString() != "")
+            if (!string.IsNullOrWhiteSpace(moduleSettings["descriptionLength"].ToString()))
             {
                 int.TryParse(moduleSettings["descriptionLength"].ToString(), out descriptionLenght);
```
Comment (david-poindexter) [inline Components/Business/AnnouncementsController.cs:283]:
```suggestion
                int.TryParse(moduleSettings["descriptionLength"].ToString(), out descriptionLength);
```

Comment (david-poindexter) [inline Components/Business/AnnouncementsController.cs:283]:
```suggestion
                int.TryParse(moduleSettings["descriptionLength"].ToString(), out descriptionLength);
```

File: Components/Business/AnnouncementsController.cs [hunk 7]
```diff
@@ -255,7 +278,7 @@ public override IList<SearchDocument> GetModifiedSearchDocuments(ModuleInfo modu
...
                 if (descriptionLenght < 1) { descriptionLenght = 1950; }
```
Comment (david-poindexter) [inline Components/Business/AnnouncementsController.cs:284]:
```suggestion
                if (descriptionLength < 1) { descriptionLength = 1950; }                    
```

File: Components/Business/AnnouncementsController.cs [hunk 8]
```diff
@@ -269,9 +292,9 @@ public override IList<SearchDocument> GetModifiedSearchDocuments(ModuleInfo modu
                 document.AuthorUserId = announcement.CreatedByUserID;
                 document.Body = announcement.Description;
                 document.CultureCode = moduleInfo.CultureCode;
-                document.Description = announcement.Description;
+                document.Description = HtmlUtils.Shorten(announcement.Description, descriptionLenght, "...");
```
Comment (david-poindexter) [inline Components/Business/AnnouncementsController.cs:295]:
```suggestion
                document.Description = HtmlUtils.Shorten(announcement.Description, descriptionLength, "...");
```

**Review by david-poindexter — CHANGES_REQUESTED** [review]
Looks good @valadas - I'm sure this was fun to track down!  😆

File: AnnouncementsEdit.ascx.cs (hunk 1 above)
Comment (mitchelsellers) [reply AnnouncementsEdit.ascx.cs:221]: My bad, looks good

**Review by mitchelsellers — APPROVED** [review]

**Comment by valadas** [comment]
> Looks good @valadas - I'm sure this was fun to track down! 😆

Yeah, it was real fun, I love timezones (NOT), why could'nt we all just use the same time, I already wake up at 2pm and go to bed at 5 am anyway...

**Pull request merged**